src/datasim_ui/handlers.cljs

```diff
@@ -84,6 +86,44 @@
    (update-in db [::db/options :options/visible]
               not)))
 
+(re-frame/reg-event-db
+  :validation/show
+  global-interceptors
+  (fn [db _]
+    (assoc-in db [::db/validation :validation/visible]
+              true)))
+
+(re-frame/reg-event-db
+ :validation/hide
+ global-interceptors
+ (fn [db _]
+   (assoc-in db [::db/validation :validation/visible]
+             false)))
+
+(re-frame/reg-event-db
+ :validation/data
+ global-interceptors
+ (fn [db [_ data]]
+   (assoc-in db [::db/validation :validation/data]
+             (map (fn [error]
+                    (merge {:visible false
+                            :id (random-uuid)}
+                           error))
+                  data))))
+
+(re-frame/reg-event-db
+ :validation/toggle-error
+ global-interceptors
+ (fn [db [_ error-id]]
+   (assoc-in db [::db/validation :validation/data]
+             (map (fn [error]
+                    (if (= error-id (:id error))
+                      (merge error {:visible
+                                    (not (:visible error))})
+                      error))
+                  (get-in db [::db/validation
+                              :validation/data])))))
+
```

Comment:
This would be more idiomatic as like `(update-in db [...] (fn [data] ...` but it's not a big deal

Reply:
done